content/en/docs/performance-workflow.md

```diff
@@ -6,11 +6,103 @@ description: >-
      This section outlines how to implement tested performance improvements.
 ---
 
-## Heading
+## Performance Workflow
 
-Edit this template to create your new page.
+This page explains the process used for performance upgrades. The four main steps to this workflow are:
+
+1. [Run MiniProfiler](#miniprofiler) on the page, look at the times for code execution and the number of queries
+1. [Extract non-performant code into it’s own method and test](#extract-and-test) it as is
+1. [Create new version of the method](#improve-performance) with the more faster code
+1. [Archive](#archive) the old code including the commit id
+
+### MiniProfiler
+
+To identify any slow and non performant code using a Profiling tool is useful. Currently MiniProfiler is included to the development server and can easily be used on any slow pages.  
+You will be able to open the MiniProfiler tool on the top left of the page in question.
+  
+<img width="343" alt="Screenshot 2021-04-16 at 09 57 29" src="https://user-images.githubusercontent.com/33180547/115031492-81026e00-9ec8-11eb-9108-f0b757735f3a.png"> 
+
+Once you have opened the MiniProfiler look for code that takes a long time to run and/or excessive numbers of SQL requests.
+  
+<img width="646" alt="Screenshot 2021-04-16 at 09 57 45" src="https://user-images.githubusercontent.com/33180547/115031984-16056700-9ec9-11eb-925c-40382dbd4021.png">
+  
+By clicking on the SQL query breakdown you can further understand which queries take long times to execute or that are repeated multiple times.
+  
+<img width="1351" alt="Screenshot 2021-04-16 at 09 59 40" src="https://user-images.githubusercontent.com/33180547/115032376-8613ed00-9ec9-11eb-8e02-06a2966951f9.png">
+  
+<img width="1264" alt="Screenshot 2021-04-16 at 10 11 26" src="https://user-images.githubusercontent.com/33180547/115032529-b3f93180-9ec9-11eb-9312-f4f4e7732735.png">
+  
+You will see that the queries have a the traceback which will lead you to the exact section of code causing the issue.  
+Here the issue is due to the code in `app/controllers/listworkers_controller.rb` :
+```
+@p_cat_choosen = Array.new
+@tab_cat.each do |category_id|
+  unless category_id.blank?
+    s_category = SCategory.find(category_id)
+    @p_cat_choosen << s_category.p_category unless @p_cat_choosen.include?(s_category.p_category)
+  end
+end
+```
+
+### Extract and Test
+
+Once you have identified the faulty code you should extract it into its own method.  
```

Comment:
This process is famously called "extract method" in the refactoring book by Martin Fowler. Maybe mention add a link or mention it to help someone Google for ways to learn how to do it in detail

Here's an example https://refactoring.com/catalog/extractFunction.html